Server/MirEnvir/Envir.cs

```diff
@@ -2532,7 +2533,93 @@ public bool CharacterExists(string name)
+                        list.Add(CharacterList[i]);
+                }
+            }
+
+            return list;
+        }
+        public List<CharacterInfo> MatchPlayerbyItem(string ItemID, bool match = false)
+        {
+            if (string.IsNullOrEmpty(ItemID)) return new List<CharacterInfo>(CharacterList);
+
+            ulong itemId = 0;
+            ulong.TryParse(ItemID, out itemId);
+
+            List<CharacterInfo> list = new List<CharacterInfo>();
+
+            for (int i = 0; i < CharacterList.Count; i++)
+            {
+                if (match)
+                {
+                    foreach (var item in CharacterList[i].Inventory)
+                        if (item != null && item.UniqueID == itemId && !list.Contains(CharacterList[i]))
+                            list.Add(CharacterList[i]);
+
+                    foreach (var item in CharacterList[i].AccountInfo.Storage)
+                        if (item != null && item.UniqueID == itemId && !list.Contains(CharacterList[i]))
+                            list.Add(CharacterList[i]);
+
+                    foreach (var item in CharacterList[i].QuestInventory)
+                        if (item != null && item.UniqueID == itemId && !list.Contains(CharacterList[i]))
+                            list.Add(CharacterList[i]);
+
+                    foreach (var item in CharacterList[i].Equipment)
+                        if (item != null && item.UniqueID == itemId && !list.Contains(CharacterList[i]))
+                            list.Add(CharacterList[i]);
+
+                    foreach (var mail in CharacterList[i].Mail)
+                        foreach (var item in mail.Items)
+                            if (item != null && item.UniqueID == itemId && !list.Contains(CharacterList[i]))
+                                list.Add(CharacterList[i]);
+                }
+                else
+                {
+                    foreach (var item in CharacterList[i].Inventory)
+                        if (item != null && item.UniqueID == itemId && !list.Contains(CharacterList[i]))
+                            list.Add(CharacterList[i]);
+
+                    foreach (var item in CharacterList[i].QuestInventory)
+                        if (item != null && item.UniqueID == itemId && !list.Contains(CharacterList[i]))
+                            list.Add(CharacterList[i]);
+
+                    foreach (var item in CharacterList[i].Equipment)
+                        if (item != null && item.UniqueID == itemId && !list.Contains(CharacterList[i]))
+                            list.Add(CharacterList[i]);
 
+                    foreach (var item in CharacterList[i].AccountInfo.Storage)
+                        if (item != null && item.UniqueID == itemId && !list.Contains(CharacterList[i]))
+                            list.Add(CharacterList[i]);
+                }
+
+            }
+
+            return list;
+        }
+        public AccountInfo GetAccountArchived(string accountID)
```

Comment:
method isn't called by anything?

Reply:
Removed

Reply:
The archived list at the top also needs removing please as that seems to have no use

Reply:
Fixed